ledger-core/virtual/integration/delegationtoken_test.go

```diff
@@ -301,83 +301,116 @@ func TestDelegationToken_IsMessageFromVirtualLegitimate(t *testing.T) {
-
-	delegationToken := server.DelegationToken(reference.NewRecordOf(class, outgoing.GetLocal()), executorRef, outgoing)
-
-	server.IncrementPulse(ctx)
-
-	typedChecker := server.PublisherMock.SetTypedChecker(ctx, mc, server)
-	typedChecker.VDelegatedCallResponse.Set(func(response *payload.VDelegatedCallResponse) bool {
-		assert.Equal(t, class, response.Callee)
-		assert.Equal(t, class, response.ResponseDelegationSpec.Callee)
-		assert.Equal(t, approver, response.ResponseDelegationSpec.Approver)
-		assert.Equal(t, outgoing, response.ResponseDelegationSpec.Outgoing)
-		return false
-	})
-
-	statePl := payload.VStateReport{
-		Status:                      payload.Empty,
-		Object:                      class,
-		OrderedPendingCount:         1,
-		OrderedPendingEarliestPulse: firstPulse.PulseNumber,
-	}
-	server.SendMessage(ctx, utils.NewRequestWrapper(server.GetPulse().PulseNumber, &statePl).SetSender(approver).Finalize())
-	server.WaitActiveThenIdleConveyor()
-
-	pl := payload.VDelegatedCallRequest{
-		Callee:         class,
-		CallFlags:      payload.BuildCallFlags(contract.CallTolerable, contract.CallDirty),
-		CallOutgoing:   outgoing,
-		DelegationSpec: delegationToken,
-	}
-	msg := utils.NewRequestWrapper(server.GetPulse().PulseNumber, &pl).SetSender(executorRef).Finalize()
-	server.SendMessage(ctx, msg)
+			logger := utils.InterceptLog(inslogger.FromContext(ctx), logHandler)
+			server.OverrideConveyorFactoryLogContext(inslogger.SetLogger(ctx, logger))
 
-	testutils.WaitSignalsTimed(t, 10*time.Second, executeDone)
-	testutils.WaitSignalsTimed(t, 10*time.Second, server.Journal.WaitAllAsyncCallsDone())
+			server.Init(ctx)
+
+			var (
+				class       = gen.UniqueGlobalRef()
+				outgoing    = server.BuildRandomOutgoingWithPulse()
+				executorRef = server.RandomGlobalWithPulse()
+				firstPulse  = server.GetPulse()
 
-	assert.False(t, errorFound)
-	assert.Equal(t, 1, typedChecker.VDelegatedCallResponse.Count())
+				delegationToken, expectedToken      payload.CallDelegationToken
+				expectedVDelegatedCallResponseCount int
+			)
+
+			server.IncrementPulse(ctx)
+
+			approver := server.RandomGlobalWithPulse()
+			jetCoordinatorMock.
+				MeMock.Return(approver).
+				QueryRoleMock.Return([]reference.Global{approver}, nil)
 
-	mc.Finish()
+			if test.haveCorrectDT {
+				delegationToken = server.DelegationToken(reference.NewRecordOf(class, outgoing.GetLocal()), executorRef, class)
+			}
+
+			server.IncrementPulse(ctx)
+
+			if test.haveCorrectDT {
+				expectedToken = server.DelegationToken(outgoing, executorRef, class)
+			}
+
+			typedChecker := server.PublisherMock.SetTypedChecker(ctx, mc, server)
+			typedChecker.VDelegatedCallResponse.Set(func(response *payload.VDelegatedCallResponse) bool {
+				assert.Equal(t, class, response.Callee)
+				assert.Equal(t, expectedToken, response.ResponseDelegationSpec)
+				assert.NotEqual(t, delegationToken.PulseNumber, response.ResponseDelegationSpec.PulseNumber)
```

Comment:
you've already checked whole response.ResponseDelegationSpec

Reply:
I think that 2 additional checks for PulseNumber and DelegateTo will not be superfluous

Reply:
hm, what purpose for these checks, if you've already checked whole response.ResponseDelegationSpec?

Reply:
fix